internal/engine/execution/procedure.go

```diff
@@ -71,173 +67,128 @@ type preparedAction struct {
-			}
+				return nil
+			}))
+		}
 
-			i := &callMethod{
-				Namespace: strings.ToLower(stmt.Extension),
-				Method:    strings.ToLower(stmt.Method),
-				Args:      args,
-				Receivers: receivers,
-			}
-			instructions = append(instructions, i)
-		case *actparser.DMLStmt:
-			// apply schema to db name in statement
-			deterministic, err := sqlanalyzer.ApplyRules(stmt.Statement, sqlanalyzer.AllRules,
-				schema, dbidSchema(schema.DBID()))
-			if err != nil {
-				return nil, err
-			}
+		for _, parsedStmt := range analyzedAction.Statements {
+			switch stmt := parsedStmt.(type) {
+			default:
+				return nil, fmt.Errorf("unknown statement type %T", stmt)
+			case *actions.ExtensionCall:
+				args, err := makeExecutables(stmt.Params)
+				if err != nil {
+					return nil, err
+				}
 
-			if deterministic.Mutative && isViewProcedure {
-				return nil, fmt.Errorf("view procedure cannot contain mutative statements")
-			}
+				i := &callMethod{
+					Namespace: stmt.Extension,
+					Method:    stmt.Method,
+					Args:      args,
+					Receivers: stmt.Receivers,
+				}
+				instructions = append(instructions, i)
+			case *actions.SQLStatement:
+				if stmt.Mutative && analyzedAction.IsView {
+					return nil, fmt.Errorf("view procedure cannot contain mutative statements")
+				}
 
-			i := &dmlStmt{
-				SQLStatement:      deterministic.Statement,
-				OrderedParameters: deterministic.ParameterOrder,
-			}
-			instructions = append(instructions, i)
-		case *actparser.ActionCallStmt:
-			args, err := makeExecutables(stmt.Args, schema)
-			if err != nil {
-				return nil, err
-			}
+				i := &dmlStmt{
+					SQLStatement:      stmt.Statement,
+					OrderedParameters: stmt.ParameterOrder,
+				}
+				instructions = append(instructions, i)
+			case *actions.ActionCall:
+				args, err := makeExecutables(stmt.Params)
+				if err != nil {
+					return nil, err
+				}
 
-			receivers := make([]string, len(stmt.Receivers))
-			for i, receiver := range stmt.Receivers {
-				receivers[i] = strings.ToLower(receiver)
-			}
+				// we must check if it is calling a view procedure or not
+				callingViewProcedure := false // callingViewProcedure tracks whether the called procedure is a view
 
-			// if calling external procedure, the procedure must be public and view.
-			// if calling internal procedure, the procedure must be view.
-			callingViewProcedure := false // callingViewProcedure tracks whether the called procedure is a view
-			if stmt.Database == schema.DBID() || stmt.Database == "" {
-				// internal
-				var action *types.Action
+				var calledAction *types.Action
 				for _, p := range schema.Actions {
-					if p.Name == stmt.Method {
-						action = p
+					if p.Name == stmt.Action {
+						calledAction = p
 						break
 					}
 				}
-				if action == nil {
-					return nil, fmt.Errorf(`procedure "%s" not found`, stmt.Method)
-				}
 
-				for _, mod := range action.Modifiers {
+				for _, mod := range calledAction.Modifiers {
```

Comment:
Any way this can fail to fall the `calledAction`, leaving it nil?

Reply:
I don't think so, but I will add a check just in case